src/libs/actions/Policy.js

```diff
@@ -244,51 +243,57 @@ function removeMembers(members, policyID) {
 }
 
 /**
- * Merges the passed in login into the specified policy
+ * Adds members to the specified workspace/policyID
  *
- * @param {Array<String>} logins
+ * @param {Array<String>} memberLogins
  * @param {String} welcomeNote
  * @param {String} policyID
  */
-function invite(logins, welcomeNote, policyID) {
-    Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`, {
-        alertMessage: '',
-    });
+function addMembersToWorkspace(memberLogins, welcomeNote, policyID) {
+    const membersListKey = `${ONYXKEYS.COLLECTION.POLICY_MEMBER_LIST}${policyID}`;
+    const logins = _.map(memberLogins, memberLogin => OptionsListUtils.addSMSDomainIfPhoneNumber(memberLogin));
 
-    // Optimistically add the user to the policy
-    const newEmployeeLogins = _.map(logins, login => OptionsListUtils.addSMSDomainIfPhoneNumber(login));
-    const employeeUpdate = {};
-    _.each(newEmployeeLogins, login => employeeUpdate[login] = {});
-    Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY_MEMBER_LIST}${policyID}`, employeeUpdate);
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: membersListKey,
 
-    // Make the API call to merge the login into the policy
-    DeprecatedAPI.Policy_Employees_Merge({
+            // Convert to object with each key containing {pendingAction: ‘add’}
+            value: _.object(logins, Array(logins.length).fill({pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD})),
+        },
+    ];
+
+    const successData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: membersListKey,
+
+            // Convert to object with each key clearing pendingAction. We don’t
+            // need to remove the members since that will be handled by onClose of OfflineWithFeedback.
+            value: _.object(logins, Array(logins.length).fill({pendingAction: null})),
```

Comment:
You need to remove the errors here, no?

Reply:
Not really, how can a login here have error? Are you considering the case something got an error and was added again and hence we should remove the error?

Reply:
> Not really, how can a login here have error?

You tried something once, it did not work (so there's an error), you try again and it works.

Reply:
yup, i think you are correct. Updating.